src/test/java/array/Array2dTest.java

```diff
@@ -0,0 +1,223 @@
+        array.visit(row -> row.visit(sb::append));
+        assertEquals("012012012", sb.toString());
+    }
+
+    @Test
+    void shouldReturnInsertedValue() {
+        // given
+        int index = 2;
+        int value = 18;
+        int size = 10;
+        Array2d<Integer> array = Array2d.of(size, size);
+        // when
+        int insertedValue = array.insert(Coordinates2d.of(index, index), value);
+        // then
+        assertEquals(value, insertedValue);
+    }
+
+    @Test
+    void shouldThrowExceptionCellIsAlreadyOccupied() {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array2d<Integer> array = Array2d.of(size, size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates2d.of(i, i),
+                        10 * i));
+        // expect
+        assertThrows(IllegalArgumentException.class,
+                () -> array.insert(Coordinates2d.of(1, 1), 100));
+    }
+
+    @Test
+    void shouldThrowAnExceptionIfIndexOutOfBound() {
+        // given
+        int size = 10;
+        Array2d<Integer> array = Array2d.of(size, size);
+        // expect
+        assertThrows(ArrayIndexOutOfBoundsException.class,
+                () -> array.insert(Coordinates2d.of(2 * size, size), 120));
+    }
+
+
+    @Test
+    void shouldVisitNonEmptyArray() {
+        // given
+        int start = 0;
+        int end = 4;
+        int size = 5;
+        Array2d<Integer> array = Array2d.of(size, size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates2d.of(i, i),
+                        10 * i));
+        String expectedPresentation = "0, null, null, null, null, \n" +
+                                      "null, 1, null, null, null, \n" +
+                                      "null, null, 2, null, null, \n" +
+                                      "null, null, null, 3, null, \n" +
+                                      "null, null, null, null, null, \n";
+        StringBuilder sb = new StringBuilder();
+        // when
+        array.visit(row -> {
+            row.visit(cell -> sb.append(cell).append(", "));
+            sb.append("\n");
+        });
+        // then
+        String presentation = sb.toString();
+        assertEquals(expectedPresentation, presentation);
+    }
+
+    @ParameterizedTest
+    @CsvSource({"1, 10", "5, 50"})
+    void shouldGiveCellValueByIndex(Integer index, Integer expectedValue) {
+        // given
+        int start = 0;
+        int end = 9;
+        int size = 10;
+        Array2d<Integer> array = Array2d.of(size, size);
+        IntStream.range(start, end)
+                .forEach(i -> array.insert(
+                        Coordinates2d.of(i, i),
+                        10 * i));
+        // when
+        Optional<Integer> value = array.at(Coordinates2d.of(index, index));
+        // then
+        value.ifPresent(
```

Comment:
I've got the same suggestion as for 1d case

Reply:
Fixed 014da09b2f32677c9fd5690d11cd3210f0d08ab1 but without exception's text message